flights/backend/services/flight_service.go

```diff
@@ -70,3 +71,65 @@ func GetFlightById(id string) (*models.Flight, error) {
 	return &flight, nil
 
 }
+
+func BuyTicket(flightId string, userId string, numOfTicketsOptional ...int) error { //numOfTickets je opcioni
+	numOfTickets := 1                  //default vrednost
+	if len(numOfTicketsOptional) > 0 { //hendlovanje opcionog parametra
+		numOfTickets = numOfTicketsOptional[0]
+	}
+
+	flight, err := GetFlightById(flightId)
+	if err != nil {
+		return err //nije nasao flight sa tim Id-em
+	}
+	if flight.Seats-numOfTickets < 0 {
+		return err //nema dovoljno slobodnih mesta
+	}
+
+	flightObjId, _ := primitive.ObjectIDFromHex(flightId)
+
+	//smanji broj slobodnih mesta za taj let
+	result, err := repos.FlightsCollection.UpdateOne(context.TODO(), bson.D{{"_id", flightObjId}}, bson.D{{"$set", bson.D{{"seats", flight.Seats - numOfTickets}}}})
```

Comment:
Maybe extract this to a separate function in `repo`, place that function in a new file `ticket_repo.go`